src/sidebar/app/utils/utils.js

```diff
@@ -11,4 +11,45 @@ function formatFooterTime(date) {
   });
 }
 
-export { formatFooterTime };
+/**
+ *
+ * @param {HTMLElement Object} parentElement
+ * @returns {HTMLElement Object}
+ */
+function getFirstNonEmptyElement(parentElement) {
+  // create an Array from parentElement's `children` (HTMLCollection)
+  let parentElementChildrenArray = [];
+  if (parentElement.children.length > 20) {
+    parentElementChildrenArray = Array.prototype.slice.call(parentElement.children[0, 20]);
+  } else {
+
+    parentElementChildrenArray = Array.prototype.slice.call(parentElement.children);
+  }
+  // search for first child element that is not empty and return it
+  for (const childElement of parentElementChildrenArray.entries()) {
```

Comment:
this can be cleaner, could probably use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find  ?